packages/react-components/src/field/src/FieldContext.tsx

```diff
@@ -0,0 +1,160 @@
+import { ElementType, ReactNode, createContext, useContext } from "react";
+import { cssModule, normalizeSize } from "../../shared";
+import { isNil } from "lodash";
+
+export interface CommonFieldContextType {
+    inputId?: string;
+    labelId?: string;
+    messageId?: string
+    required?: boolean;
+    disabled?: boolean;
+    size?: "sm" | "md";
+    fluid?: boolean;
+    validationState?: "valid" | "invalid";
+    hasLabel?: boolean;
+    hasMessage?: boolean;
+    labelClassName?: string;
+    inputClassName?: string;
+    messageClassName?: string;
+}
+
+export interface FieldContextType extends CommonFieldContextType {
+    isGroup?: boolean;
+}
+
+export interface UseFieldContextType extends CommonFieldContextType {
+    isGroupField?: boolean;
+}
+
+export const FieldContext = createContext<FieldContextType>(null);
+
+export function useFieldContext(): [UseFieldContextType, boolean] {
+    const context = useContext(FieldContext);
+
+    if (!isNil(context)) {
+        const { isGroup: isGroupField = false, ...rest } = context;
```

Comment:
Oh well my renaming was cute without TS. Now it seems stupid haha. Maybe we should normalize this to `isGroup` ?

Reply:
Haha agreed, i renamed everything and it simplified the typings a lot